cmd/system.go

```diff
@@ -43,10 +51,35 @@ var systemCmd = &cobra.Command{
 var systemDeployCmd = &cobra.Command{
 	Use:   "deploy",
 	Short: "Deploy DCE to a new master account",
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Before running the command, resolve the configuration by using
+		// the default load order.
+		deployConfig.TFInitOptions = *cfg.Coalesce(&deployConfig.TFInitOptions, Config.Terraform.TFInitOptions, stringp(DCETFInitOptionsEnvVar), stringp(Empty))
+		deployConfig.TFApplyOptions = *cfg.Coalesce(&deployConfig.TFApplyOptions, Config.Terraform.TFApplyOptions, stringp(DCETFApplyOptionsEnvVar), stringp(Empty))
+
 		ctx := context.WithValue(context.Background(), constants.DeployConfig, &deployConfig)
 		if err := Service.Deploy(ctx, &deployOverrides); err != nil {
-			log.Fatalln(err)
+			return err
 		}
+		return nil
 	},
+	// If the command was successful, read the config file back in
+	// and if it's different than the running config, then save it.
+	// We don't want to do this unless the command was succesful,
+	// though, because of cases like bad tf opts we don't want to
+	// create an usuable state.
```

Comment:
Not to rehash too much, but I'm kind of wondering if this makes sense for end users, to persist CLI flags to dce.yaml.
To make an analogy to another tool, imagine if you were deploying some app you were working on locally, and called

`terraform apply -var namespace=nathan`

...and it resulted in terraform writing to a local `deploy.auto.tfvars` file with `namespace = "nathan"`, so that every future call to `terraform apply` used the same vars by  default.

It seems to me that the point of having CLI flags is to be able to _temporarily_ override your default configuration that's writen your file system or git repo or whatever, without persisting those changes.

To make it more concrete -- at some point, we'll want our DCE @ Optum repo to use the dce-cli. We will probably commit our `dce.yaml` file to the repo, to define as-code how our environment is configured. But individual developers may want to deploy locally using a different `namespace`, or DCE version, or whatever, without committing the change to the repo.


Anyways... I'll leave the decision up to you, but those are my two cents

Reply:
Per our sidebar, a `--save-options` saves them to the config file and is default false. If someone is an advanced terraform user, they should look into using the terraform module. The CLI is for those that aren't.